api/src/shared/infrastructure/repositories/challenge-repository.js

```diff
@@ -122,38 +120,29 @@ export async function findValidatedBySkills(skills, locale) {
 }
 
 export async function findActiveFlashCompatible({
-  date,
   locale,
   successProbabilityThreshold = config.features.successProbabilityThreshold,
   accessibilityAdjustmentNeeded = false,
-  complementaryCertificationKey,
+  version,
   dependencies = {
     getInstance,
   },
 } = {}) {
   _assertLocaleIsDefined(locale);
-
-  const hasCoreScope =
-    !complementaryCertificationKey || complementaryCertificationKey === ComplementaryCertificationKeys.CLEA;
-
-  const cacheKey = `findActiveFlashCompatible({ scope: ${hasCoreScope ? Frameworks.CORE : complementaryCertificationKey}, locale: ${locale}, accessibilityAdjustmentNeeded: ${accessibilityAdjustmentNeeded}})`;
-
+  const cacheKey = `findActiveFlashCompatible({ versionId: ${version?.id}, locale: ${locale}, accessibilityAdjustmentNeeded: ${accessibilityAdjustmentNeeded} })`;
```

Comment:
Possible que version soit null ? Ha dans le cas de CORE peut etre ? Si oui je propose de le montrer clairement pour se souvenir que faut enlever l'indication du null possible

```suggestion
  const versionCached = version ? version.id : Frameworks.CORE;
  const cacheKey = `findActiveFlashCompatible({ versionId: ${versionCached}, locale: ${locale}, accessibilityAdjustmentNeeded: ${accessibilityAdjustmentNeeded} })`;
```

Si le null est pas possible alors l'operator `?` peut sauter

```suggestion
  const cacheKey = `findActiveFlashCompatible({ versionId: ${version.id}, locale: ${locale}, accessibilityAdjustmentNeeded: ${accessibilityAdjustmentNeeded} })`;
```

Reply:
Oui c'est possible que ça soit null dans le cas de Core tant que l'on n'a pas changé la méthode dans le repo pour qu'elle soit unique aux deux types de certification.
Du coup je trouverais étrange d'avoir un ID dans le cas où la version existe et un scope dans le cas où cela n'existe pas.
Je propose de régler ce souci lorsqu'on unifiera la méthode.